File: tree.cpp
#include <bits/stdc++.h>
using namespace std;

struct Node {
    int data;
    struct Node* left;
    struct Node* right;

    Node(int val)
    {
        data = val;
        left = NULL;
        right = NULL;
    }
};

int main(){
    /*
             0
           /  \
          1    2
         / \  / \
        3  4  5  6
    */
    struct Node* root = new Node(0);
    root->left = new Node(1);
    root->right = new Node(2);
    root->left->left = new Node(3);
    root->left->right = new Node(4);
    root->right->left = new Node(5);
    root->right->right = new Node(6);

    return 0;
}

File: tree.py
class Node(object):
    def __init__(self, data):
        self.val = data
        self.left = None
        self.right = None


def func(root):
    pass


def return_tree():
    """
    Returns the root of a tree as shown below:
            0
           /  \
          1    2
         / \  / \
        3  4  5  6
    :return: Reference to root Node
    """
    root = Node(0)
    root.left = Node(1)
    root.right = Node(2)
    root.left.left = Node(3)
    root.left.right = Node(4)
    root.right.left = Node(5)
    root.right.right = Node(6)
    return root

